packages/collections/_test.pony

```diff
@@ -752,3 +753,158 @@ class iso _TestHeap is UnitTest
+    let set0011: _TestFlagsFlags = _TestFlagsFlags
+                                   .>set(_TestFlagB)
+                                   .>set(_TestFlagA)
+    h.assert_eq[U8](set0011.value(), 0b0011)
+
+
+    let set0101: _TestFlagsFlags = _TestFlagsFlags
+                                   .>set(_TestFlagC)
+                                   .>set(_TestFlagA)
+    h.assert_eq[U8](set0101.value(), 0b0101)
+
+    // Immutable Bitwise Operators: op_or, op_and, op_xor
+                                                  // 0b0011
+                                                  // 0b0101 or
+    h.assert_eq[U8](set0011.op_or(set0101).value(),  0b0111)
+
+                                                  // 0b0011
+                                                  // 0b0101 and
+    h.assert_eq[U8](set0011.op_and(set0101).value(), 0b0001)
+
+                                                  // 0b0011
+                                                  // 0b0101 xor
+    h.assert_eq[U8](set0011.op_xor(set0101).value(), 0b0110)
+
+    // Immutable Bitwise Operators: add, sub
+                                               // 0b0011
+                                               // 0b1000 add
+    h.assert_eq[U8]((set0011+_TestFlagD).value(), 0b1011)
+
+                                               // 0b0011
+                                               // 0b0001 sub
+    h.assert_eq[U8]((set0011-_TestFlagA).value(), 0b0010)
+
+    // Mutable Bitwise Operator: without
+                                                   // 0b0011
+                                                   // 0b0101 without
+    h.assert_eq[U8](set0011.without(set0101).value(), 0b0010)
+                                                   // 0b0101
+                                                   // 0b0011 without
+    h.assert_eq[U8](set0101.without(set0011).value(), 0b0100)
+
+    // Mutable Operator: union
+    t = set0011.clone()      //0b0011
+    t.union(set0101)         //0b0101
+    h.assert_eq[U8](t.value(), 0b0111)
+
+    // a.union(b) == b.union(a)
+    t = set0101.clone()      //0b0101
+    t.union(set0011)         //0b0011
+    h.assert_eq[U8](t.value(), 0b0111)
+
+    // Mutable Operator: intersect
+    t = set0011.clone()      //0b0011
+    t.intersect(set0101)     //0b0101
+    h.assert_eq[U8](t.value(), 0b0001)
+
+    // a.intersect(b) == b.intersect(a)
+    t = set0101.clone()      //0b0101
+    t.intersect(set0011)     //0b0011
+    h.assert_eq[U8](t.value(), 0b0001)
+
+    // Mutable Operator: difference
+    t = set0011.clone()              //0b0011
+    t.difference(set0101)            //0b0101
+    h.assert_eq[U8](t.value(), 0b0110)
+
+    // a.difference(b) == b.difference(a) **symmetric difference**
+    t = set0101.clone()      //0b0101
+    t.difference(set0011)    //0b0011
+    h.assert_eq[U8](t.value(), 0b0110)
+
+    // Mutable Operator: remove
+    t = set0011.clone()      //0b0011
+    t.remove(set0101)        //0b0101
```

Comment:
you are removing these comments as well, yes?

Reply:
and the other equivs..., yes?

Reply:
I wasn't planning to.

The intent was to show the bitmasks aligned vertically so it was obvious to the reader which operation was being performed:

```
    // Mutable Operator: remove
    t = set0011.clone()      //0b0011
    t.remove(set0101)        //0b0101
    h.assert_eq[U8](t.value(), 0b0010)
```

It just looks different from other examples:
```
    // Mutable Bitwise Operator: without
                                                   // 0b0011
                                                   // 0b0101 without
    h.assert_eq[U8](set0011.without(set0101).value(), 0b0010)
                                                   // 0b0101
                                                   // 0b0011 without
    h.assert_eq[U8](set0101.without(set0011).value(), 0b0100)
```

I planned on leaving them in as that clarity was how I found the bug in remove() in the first place.

Reply:
I personally dont think they add any value and at best are noise, but if you want to leave them, ill go with that and they can be merged for now and others can weigh in later. given the names you have, i really think it is duplicating information that is already there and in the process, it makes it harder to me to absorb what is going on.

Reply:
The intent probably would have been clearer had I added the operation there too, like this:

```
    // Mutable Operator: remove
    t = set0011.clone()      //0b0011
    t.remove(set0101)        //0b0101 remove
    h.assert_eq[U8](t.value(), 0b0010)
```

Reply:
(Now I'm spotting the missing space between the // and the 0b and wanting to make those corrections too)

I can remove the comment if you think it's superfluous.  But as someone for who bitwise math isn't second nature having the values vertically aligned made it a lot easier for me to reason about.

Reply:
I think that you wrote the tests in such a way that they really arent needed

`set0011` equates very well to `0b0011` and in my mind, reducing the comment to noise. ditto for the other comments as well. but, i will defer to you on this.

feel free to do any additional clean up you want then drop me a note when its time for a final look and merge.

Reply:
It's really interesting to me that what I find essential you find noise that makes it more difficult to absorb.

Brains are interesting things.

Reply:
It might be less noisy to me if it was on the line above rather than "to the side".

Reply:
Like this?

```
    // Mutable Operator: remove
    //  0b0011
    //  0b0101 remove
    //  0b0010
    t = set0011.clone()
    t.remove(set0101)
    h.assert_eq[U8](t.value(), 0b0010)
```

Reply:
Like:

```pony
    // Mutable Operator: remove
    //  start with 0b0011
    t = set0011.clone()
    //  remove 0b0101
    t.remove(set0101)
    // should be 0b0010
    h.assert_eq[U8](t.value(), 0b0010)
```

Reply:
or...
```
    // Mutable Operator: remove
    t = set0011.clone()     // 0b0011
    t.remove(set0101)       // 0b0101 remove
                            // 0b0010
    h.assert_eq[U8](t.value(), 0b0010)
```

Reply:
that is just noise to me, your 2nd one.

Reply:
I don't like the second one either, it's duplication of a value into a comment which could get out of sync with the line below.

For what I'm using it for (Looking at the two input bitstrings and doing a column-by-column verification in my head) the version you have wouldn't work for me.  I need the numbers aligned for my brain to keep track.

Still, I'm sure we all agree that my first version would be unacceptable to everyone:

```
                      t = set0011.clone()
                 t.remove(set0101)
h.assert_eq[U8](t.value(), 0b0010)
```

I'm flexible regardless.

Reply:
Yes, that wouldn't be in line with the style guide and is also just awful. 😉 

I definitely find the current way you have it to be very distracting. However, as I said, I'm good with merging it as it is. It is an improvement and other can weigh in later.

Reply:
Sounds good!

Thank you for your help and guidance today.  I appreciate it.